src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandHandler.cs

```diff
@@ -48,29 +52,38 @@ public async Task<List<PSCommandMessage>> Handle(GetCommandParams request, Cance
             PSCommand psCommand = new PSCommand();
 
             // Executes the following:
-            // Get-Command -CommandType Function,Cmdlet,ExternalScript | Select-Object -Property Name,ModuleName | Sort-Object -Property Name
+            // Get-Command -CommandType Function,Cmdlet,ExternalScript | Sort-Object -Property Name
             psCommand
                 .AddCommand("Microsoft.PowerShell.Core\\Get-Command")
                     .AddParameter("CommandType", new[] { "Function", "Cmdlet", "ExternalScript" })
-                .AddCommand("Microsoft.PowerShell.Utility\\Select-Object")
-                    .AddParameter("Property", new[] { "Name", "ModuleName" })
                 .AddCommand("Microsoft.PowerShell.Utility\\Sort-Object")
                     .AddParameter("Property", "Name");
 
-            IEnumerable<PSObject> result = await _powerShellContextService.ExecuteCommandAsync<PSObject>(psCommand).ConfigureAwait(false);
+            IEnumerable<CommandInfo> result = await _powerShellContextService.ExecuteCommandAsync<CommandInfo>(psCommand).ConfigureAwait(false);
 
             var commandList = new List<PSCommandMessage>();
             if (result != null)
             {
-                foreach (dynamic command in result)
+                foreach (CommandInfo command in result)
                 {
+                    // Get the default ParameterSet
+                    string defaultParameterSet = null;
+                    foreach (CommandParameterSetInfo parameterSetInfo in command.ParameterSets)
+                    {
+                        if (parameterSetInfo.IsDefault)
+                        {
+                            defaultParameterSet = parameterSetInfo.Name;
+                            break;
+                        }
+                    }
+
                     commandList.Add(new PSCommandMessage
                     {
                         Name = command.Name,
                         ModuleName = command.ModuleName,
                         Parameters = command.Parameters,
                         ParameterSets = command.ParameterSets,
-                        DefaultParameterSet = command.DefaultParameterSet
+                        DefaultParameterSet = defaultParameterSet
                     });
                 }
```

Comment:
I left this cause it's cleaner... but it really has nothing to do with this PR...

Reply:
Wait, what's the context here? Is there an issue with `command.DefaultParmeterSet`?

Reply:
IIRC it's a NoteProperty and doesn't actually exist